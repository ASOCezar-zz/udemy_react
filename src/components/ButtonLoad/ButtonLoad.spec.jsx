import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ButtonLoad } from '.';

describe('<Button />', () => {
  it('should render one text "Load More" on button', () => {
    const fn = jest.fn();
    render(<ButtonLoad text="Load More" onClick={fn} />);

    expect(screen.getByRole('button', { name: /load more/i })).toBeInTheDocument();
  });

  it('should call a function when clicked', () => {
    const fn = jest.fn();
    render(<ButtonLoad text="Load More" onClick={fn} />);

    userEvent.click(screen.getByRole('button'));

    expect(fn).toHaveBeenCalled();
  });

  it('should be disabled when disabled equal true', () => {
    const fn = jest.fn();
    render(<ButtonLoad text="Load More" disabled={true} onClick={fn} />);

    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('should be enable when disabled equal false', () => {
    const fn = jest.fn();
    render(<ButtonLoad text="Load More" disabled={false} onClick={fn} />);

    expect(screen.getByRole('button')).toBeEnabled();
  });

  it('should match with snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<ButtonLoad text="load more" disabled={false} onClick={fn} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
