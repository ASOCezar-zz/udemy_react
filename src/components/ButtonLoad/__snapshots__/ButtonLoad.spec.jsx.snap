// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match with snapshot 1`] = `
<button
  class="sc-bdnxRM RTIoa"
>
  load more
</button>
`;
